aws/credentials/process_provider.go

```diff
@@ -0,0 +1,228 @@
+	}
+
+	// Execute command
+	cmd := exec.Command(command, cmdArgs...)
+	cmd.Env = os.Environ()
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, awserr.New("ProcessProviderLoad", "Error executing credential_process", err)
+	}
+	return out, nil
+}
+
+func (p *ProcessProvider) credentialProcess(process string) (Value, error) {
+	out, err := p.executionFunc(process)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, err
+	}
+
+	// Serialize and validate response
+	resp := &credentialProcessResponse{}
+	err = json.Unmarshal(out, resp)
+	if err != nil {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "Error parsing credential_process output: "+string(out), err)
+	}
+	if resp.Version != 1 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Version in credential_process output is not 1", err)
+	}
+	if len(resp.AccessKeyID) == 0 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Missing required AccessKeyId in credential_process output", err)
+	}
+	if len(resp.SecretAccessKey) == 0 {
+		return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", " Missing required SecretAccessKey in credential_process output", err)
+	}
+
+	// Handle expiration
+	if len(resp.Expiration) > 0 {
+		expiry, err := time.Parse(time.RFC3339, resp.Expiration)
+		if err != nil {
+			return Value{ProviderName: ProcessProviderName}, awserr.New("ProcessProviderLoad", "Error parsing expiration of credential_process output: "+resp.Expiration, err)
+		}
+		p.SetExpiration(expiry, p.ExpiryWindow)
+	}
+
+	return Value{
+		ProviderName:    ProcessProviderName,
+		AccessKeyID:     resp.AccessKeyID,
+		SecretAccessKey: resp.SecretAccessKey,
+		SessionToken:    resp.SessionToken,
+	}, nil
+
+}
+
+// filename returns the filename to use to read AWS shared credentials.
+//
+// Will return an error if the user's home directory path cannot be found.
+func (p *ProcessProvider) filename() (string, error) {
+	if len(p.Filename) != 0 {
+		return p.Filename, nil
+	}
+
+	if p.Filename = os.Getenv("AWS_CONFIG_FILE"); len(p.Filename) != 0 {
+		return p.Filename, nil
+	}
+
+	if home := shareddefaults.UserHomeDir(); len(home) == 0 {
+		// Backwards compatibility of home directly not found error being returned.
+		// This error is too verbose, failure when opening the file would of been
+		// a better error to return.
+		return "", ErrProcessProviderHomeNotFound
+	}
+
+	p.Filename = shareddefaults.SharedConfigFilename()
+
+	return p.Filename, nil
+}
+
+// profile returns the AWS shared config profile. If empty will read
+// environment variable "AWS_PROFILE". If that is not set profile will
+// return "default".
+func (p *ProcessProvider) profile() string {
+	if p.Profile == "" {
+		p.Profile = os.Getenv("AWS_PROFILE")
+	}
+	if p.Profile == "" {
+		p.Profile = os.Getenv("AWS_DEFAULT_PROFILE")
```

Comment:
The SDK does not use the `AWS_DEFAULT_PROFILE` by default. this was a purely CLI environment variable that still exists for backwards compatibility.  The SDK's session and env_config. go provides the logic to determine which environment variable to use.